File: 1.FDP/3° Competencia/Java/Repeticion/Ejercicio7.java
import java.util.Scanner;	

public class Ejercicio7 {
	public static void main(String[] args){
		Scanner scan= new Scanner(System.in);
		for (int i=0; i<3; i++) {
			System.out.println("Calcular el costo de cada ingrediente.\n");
			double CantidadDeVenta, Costo1, Cantidad, Costo;
			System.out.println("Cantidad por la que se vende:");
			CantidadDeVenta= scan.nextDouble();
			if (CantidadDeVenta > 0) {
				System.out.println("Costo de venta por ingrediente por " + CantidadDeVenta + ":");
				Costo1= scan.nextDouble ();
				if (Costo1 > 0) {
					System.out.println("La cantidad a usar");
					Cantidad= scan.nextDouble ();
					if (Cantidad > 0) {
						Costo= Cantidad * Costo1 / CantidadDeVenta;
						System.out.println("Costo de ingrediente= $" + Costo);
						System.out.println("////////////////////////////////////////.");
					} else {
						System.out.println("Error en la cantidad a usar");
						System.out.println("/////////////////////////////////");
					}
				} else {
					System.out.println("Error en el de venta");
					System.out.println("//////////////////////////");
				}
			} else {
				System.out.println("Error en la cantidad de venta");
				System.out.println("////////////////////////////////////");
			}	
		}
	}
}

File: 1.FDP/3° Competencia/C++/Repeticion/Ejercicio7.cpp
#include <iostream>
using namespace std;

int main (){
	cout << "Calcular el costo por ingrediente. \n";
	double CantidadDeVenta, Costo1, Cantidad, Costo;
	string Unidad;
	int respuesta;
	cout << "Cuantas veces deceas que se repita: ";
	cin >> respuesta;
	for ( int i=0; i<respuesta; i++) {
		cout << "Ingrear unidad de medida :";
		cin >> Unidad;
		cout << "Ingresar la cantidad en la que se vende por " << Unidad << ":";
		cin >> CantidadDeVenta;
		if (CantidadDeVenta > 0) {
			cout << "Ingresar el precio de venta por " << CantidadDeVenta << "-" << Unidad << ":";
			cin >> Costo1;
			if (Costo1 > 0) {
				cout << "Ingresar la cantidad a usar:";
				cin >> Cantidad;
				if (Cantidad > 0) {
					Costo= Cantidad * Costo1 / CantidadDeVenta;
					cout << "Costo de ingrediente = $" << Costo  << "\n";
					cout << "///////////////////////////////. \n";
				} else {
					cout << "Error a cantidad a usar.\n";
					cout << "////////////////////////////.\n";
				}
			} else {
				cout << "Error en el costo de venta.\n";
				cout << "////////////////////////////////.\n";
			}
		} else {
			cout << "Eror en la cantidad de venta.\n";
			cout << "/////////////////////////////////.\n";
		}
	}	
	return 0;
}